### PR TITLE
Update gradle plugin version, buildToolsVersion, targetSdkVersion, compileSdkVersion, explicitly set useAndroidX=true, add SpotBugs tasks

### PR DESCRIPTION
**Pull request by iambmelt**
Draft PR for updating Gradle, adding SpotBugs.

Work in progress.

### Available SpotBugs Tasks:
```
common:spotbugsDebug
common:spotbugsRelease
```

### Execute SpotBugs Tasks Locally:
```
> ./gradlew spotBugsDebug // Or whichever variant you require
```

Similar:
- #1002

### CODE REVIEW

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iambmelt) [inline .idea/jarRepositories.xml:1]: This looks like it needs to be ignored....

File: gradle/versions.gradle [hunk 2]
```diff
@@ -3,9 +3,9 @@ ext {
     // SDK
     minSdkVersion = 16
     automationAppMinSDKVersion = 21
-    targetSdkVersion = 28
-    compileSdkVersion = 28
-    buildToolsVersion = "28.0.3"
+    targetSdkVersion = 29
+    compileSdkVersion = 29
+    buildToolsVersion = "29.0.3"
```
Comment (AdamBJohnsonx) [inline gradle/versions.gradle:8]: This is redundant and ignored by Gradle, you can ditch it.

Comment (iambmelt) [reply gradle/versions.gradle]: Removed in cacd757ffb514a5c905465340f948058c0c0a8d4

File: common/build.gradle [hunk 3]
```diff
@@ -144,16 +145,16 @@ android.libraryVariants.all { variant ->
 // Create tasks to generate javadoc jar
 android.libraryVariants.all { variant ->
     task("${variant.name}JavadocJar", type: Jar, dependsOn: "${variant.name}Javadoc") {
-        classifier = 'javadoc'
-        from tasks["${variant.name}Javadoc"].destinationDir
+        archiveClassifier.set("javadoc")
+        from tasks["${variant.name}Javadoc"].destinationDir // 'destinationDir' is still reqd here.
     }
 }
```
Comment (AdamBJohnsonx) [inline common/build.gradle:152]: You probably want a spot bugs configuration to set it to accept failure, and a backlog task to fix the failures.

File: .idea/jarRepositories.xml (hunk 1 above)
Comment (iambmelt) [reply .idea/jarRepositories.xml]: 153c1d473e486bdb4ef885b731f97024c919d4f6

File: common/build.gradle (hunk 3 above)
Comment (iambmelt) [reply common/build.gradle:152]: Have to do this in a few places:
- PMD: 676190a02fe33c1acd84c930fe8c6d439e7c09ae
- SpotBugs: f7eb3b5e53378d00e1498a1b7509072dadb33aeb

Comment (iambmelt) [reply common/build.gradle:152]: TODO: Create backlog items

File: common/build.gradle [hunk 4]
```diff
@@ -175,16 +176,24 @@ task checkstyle(type: Checkstyle) {
...
+    ignoreFailures = true
     reportsDir = file("$project.buildDir/outputs/")
 }
 
 task pmd(type: Pmd) {
     description 'Run pmd'
     group 'verification'
 
-    ruleSets = ["java-basic", "java-braces", "java-strings", "java-unusedcode"]
+    //ruleSets = ["java-basic", "java-braces", "java-strings", "java-unusedcode"]
```
Comment (iambmelt) [inline common/build.gradle:195]: Remove dead code

Comment (iambmelt) [reply common/build.gradle]: bc37f821951f3e98e6bed92498af737a42247a86

File: common/build.gradle [hunk 5]
```diff
@@ -175,16 +176,23 @@ task checkstyle(type: Checkstyle) {
     classpath = files()
 }
 
+spotbugs {
+    // Ignoring failures for now. TODO: create workitem
```
Comment (iambmelt) [inline common/build.gradle:180]: TODO

File: common/build.gradle [hunk 6]
```diff
@@ -175,16 +176,23 @@ task checkstyle(type: Checkstyle) {
...
+    ignoreFailures = true
+}
+
 pmd {
-    ignoreFailures = false
+    toolVersion "6.28.0"
+    // Ignoring failures for now. TODO: create workitem
```
Comment (iambmelt) [inline common/build.gradle:186]: TODO

**Pull request closed without merging**